docs/DeveloperGuide.adoc

```diff
@@ -153,94 +153,234 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+In this class, invalid arguments will result in a `ParseException`. +
+Two types of invalid arguments are checked for: empty arguments
+and arguments with only whitespace characters.
 
-image::UndoRedoState0.png[]
+The `execute()` function of the `DoneCommand` will first check if the given arguments
+are in the format required for an order to be marked as complete
+(i.e. both the command word `return` and `INDEX` are present)
+The check is done within the `execute()` function. +
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+If the input is valid, the particular order based on the integer `INDEX` will be used to +
+obtain this order from the order book. This order will be checked if it was delivered. +
 
-image::UndoRedoState1.png[]
+1. If it was not delivered, it will trigger the `setOrder()`, `deliverOrder()`
+and `updateFilteredOrderList()` functions. These methods will be used to replace
+the existing order with the new order that has a delivered status. Also, the order book
+will be updated to show the change. +
 
-Step 3. The user executes `add n/David ...` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+2. If the order was delivered, it will only trigger the `updateFilteredOrderList()` function which will
+display the original order book.
 
-image::UndoRedoState2.png[]
+=== Nearby feature
+The nearby feature allows the user to view all orders that are located at a particular area
+based on a given search criteria. +
+There are currently two possible search options:
+
+. Searching by *postal sector*
+* A *postal sector* refers to the first *two* digits of a six digit Singapore postal code.
+* The list of postal sectors and their corresponding general locations can be found
+https://www.ura.gov.sg/realEstateIIWeb/resources/misc/list_of_postal_districts.htm[on this website].
+
+[start=2]
+. Searching by *area*
+* An *area* refers to one of the five areas of Singapore
+** Central
+** East
+** North East
+** West
+** North
+* You can obtain more detailed information about each area from
+https://keylocation.sg/singapore/districts-map[this website]
+
+The nearby feature was implemented as a `NearbyCommand` in the `Logic` package. +
+
+==== Class Diagram for Nearby feature
+.Nearby Command Class Diagram
+image::NearbyClassDiagram.svg[]
+
+==== Sequence Diagram for searching by postal sector
+The sequence diagram for the *Nearby Command* when searching by *postal sector*
+is shown below:
+
+.Nearby Command Sequence Diagram
+image::NearbyCommandSequenceDiagram1.svg[]
```

Comment:
The activation bar for method invocation updateFilteredOrderList from logic to model might consider to end earlier.